src/PKSim.Core/Mappers/PKValuesToPKAnalysisMapper.cs

```diff
@@ -31,15 +31,21 @@ public class PKValuesToPKAnalysisMapper : IPKValuesToPKAnalysisMapper
          _displayUnitRetriever = displayUnitRetriever;
       }
 
-      public PKAnalysis MapFrom(DataColumn dataColumn, PKValues pkValues, PKParameterMode mode, string moleculeName)
+      public PKAnalysis MapFrom(DataColumn dataColumn, PKValues pkValues, PKParameterMode mode, string moleculeName, bool forPopulation)
       {
          var pk = new PKAnalysis().WithName(moleculeName);
-         _pkParameterRepository.All().Where(parameter => parameter.Mode.Is(mode)).Each(parameter => pk.Add(createPKParameter(parameter, pkValues)));
+         _pkParameterRepository.All().Where(parameter => parameter.Mode.Is(mode) && (!forPopulation || !filteNaNValues(parameter.Name, pkValues))).Each(parameter => pk.Add(createPKParameter(parameter, pkValues)));
```

Comment:
THis is very hard to read
` parameter.Mode.Is(mode) && (!forPopulation || !filteNaNValues(parameter.Name, pkValues)))`

I would suggest to create a method `shouldExport(parameter, pkValues, mode)` and then you can make it easy on the eyes